build_tools/benchmarks/post_benchmarks_as_pr_comment.py

```diff
@@ -366,34 +375,46 @@ def parse_arguments():
 def main(args):
   benchmark_files = expand_and_check_file_paths(args.benchmark_files)
   compile_stats_files = expand_and_check_file_paths(args.compile_stats_files)
+
+  github_token = get_required_env_var('GITHUB_TOKEN')
+  dashboard_api_url = f"{get_required_env_var('IREE_DASHBOARD_URL')}/apis/v2"
+
+  pr_number = args.pr_number
   full_md, abbr_md = get_benchmark_result_markdown(
-      benchmark_files,
-      compile_stats_files,
-      query_base=args.query_base,
+      benchmark_files=benchmark_files,
+      compile_stats_files=compile_stats_files,
       comment_title=args.comment_title,
+      pr_commit=args.pr_commit,
+      pr_url=f"{GITHUB_IREE_REPO_PREFIX}/pull/{pr_number}",
+      build_url=args.build_url,
+      base_branch=args.pr_base_branch,
+      dashboard_api_url=dashboard_api_url,
       verbose=args.verbose)
 
   if args.dry_run:
     print(full_md, "\n\n", abbr_md)
     return
 
-  pr_number = get_required_env_var("BUILDKITE_PULL_REQUEST")
-  # Buildkite sets this to "false" if not running on a PR:
-  # https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars-buildkite-pull-request
-  if pr_number == "false":
-    raise ValueError("Not a pull request")
-
-  build_number = get_required_env_var("BUILDKITE_BUILD_NUMBER")
-  filename = f"iree-full-benchmark-result-{build_number}.md"
-  gist_url = post_to_gist(filename, full_md, args.verbose)
+  filename = f"iree-full-benchmark-result-{pr_number}.md"
```

Comment:
I changed to using the PR number in the filename instead of the build number, which will need to introduce an extra argument. I think it's fine since gist allows duplicate filenames and there is a link to the build page in each post.